### PR TITLE
Adding support for (multiple) swift backends

### PR DESCRIPTION
**Pull request by mnederlof**
Add support for swift_store_config_file setting and generating the references based on pillar data.

### CODE REVIEW

**Comment by mnederlof** [comment]
This is the new way of managing swift, and also 'hides' the authentication credentials if direct url / locations are exposed via the glance API

File: tests/pillar/single_swift_references.sls [hunk 1]
```diff
@@ -0,0 +1,69 @@
+glance:
+  server:
+    enabled: true
+    version: liberty
```
Comment (pupapaik) [inline tests/pillar/single_swift_references.sls]: Can you please add this test also for mitaka to verify that your config files does not have any syntax error?

**Comment by pupapaik** [comment]
Just small adjustment with tests for mitaka to see validity of config.

**Review by pupapaik — CHANGES_REQUESTED** [review]

**Comment by mnederlof** [comment]
Updated the pillar test data

**Pull request merged**